17:35:12  isort (ote_sdk)..........................................................Passed
17:35:12  isort (ote_cli|external).................................................Passed
17:35:15  black (ote_sdk|ote_cli)..................................................Passed
17:35:16  black (rest).............................................................Passed
17:35:18  flake8...................................................................Passed
17:35:19  prettier.................................................................Passed
17:35:23  mypy (ote_sdk)...........................................................Failed
17:35:23  - hook id: mypy
17:35:23  - exit code: 1
17:35:23  
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1793: error: List item 0 has incompatible type "Optional[CurveMetric]"; expected "CurveMetric"  [list-item]
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1801: error: List item 0 has incompatible type "Optional[ScoreMetric]"; expected "Union[ScoreMetric, CountMetric, InfoMetric, DateMetric, DurationMetric]"  [list-item]
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1807: error: List item 0 has incompatible type "Optional[CurveMetric]"; expected "CurveMetric"  [list-item]
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1815: error: List item 0 has incompatible type "Optional[ScoreMetric]"; expected "Union[ScoreMetric, CountMetric, InfoMetric, DateMetric, DurationMetric]"  [list-item]
17:35:23  Found 4 errors in 1 file (checked 172 source files)
17:35:23  
17:35:24  mypy (ote_cli)...........................................................Passed
17:35:25  mypy (external)..........................................................Passed
17:36:12  pylint...................................................................Passed
17:36:12  [Pipeline] }
17:36:12  ERROR: script returned exit code 1

--- a/ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py
+++ b/ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py
@@ -13,6 +13,7 @@
 # and limitations under the License.
 
 import datetime
+from typing import cast
 
 import numpy as np
 import pytest
@@ -1742,12 +1743,7 @@ class TestFMeasure:
                         == VisualizationType.RADIAL_BAR
                     )
                 if isinstance(expected_metric_group, LineMetricsGroup):
-                    assert len(actual_metric_group.metrics) == 1
-                    expected_metric = expected_metric_group.metrics[0]
-                    actual_metric = actual_metric_group.metrics[0]
-                    assert actual_metric.name == expected_metric.name
-                    assert actual_metric.ys == expected_metric.ys
-                    assert actual_metric.xs == expected_metric.xs
+                    assert actual_metric_group.metrics == expected_metric_group.metrics
                     assert (
                         actual_metric_group.visualization_info.name
                         == expected_metric_group.visualization_info.name
@@ -1796,28 +1792,12 @@ class TestFMeasure:
         def generate_expected_optional_dashboard_metric_groups(
             actual_f_measure: FMeasure,
         ):
-            per_confidence = CurveMetric(
-                name=actual_f_measure.f_measure_per_confidence.name,  # type: ignore[union-attr]
-                ys=actual_f_measure.f_measure_per_confidence.ys,  # type: ignore[union-attr]
-                xs=actual_f_measure.f_measure_per_confidence.xs,  # type: ignore[union-attr]
-            )
-            optimal_confidence = ScoreMetric(
-                name=actual_f_measure.best_confidence_threshold.name,  # type: ignore[union-attr]
-                value=actual_f_measure.best_confidence_threshold.value,  # type: ignore[union-attr]
-            )
-            per_nms = CurveMetric(
-                name=actual_f_measure.f_measure_per_nms.name,  # type: ignore[union-attr]
-                ys=actual_f_measure.f_measure_per_nms.ys,  # type: ignore[union-attr]
-                xs=actual_f_measure.f_measure_per_nms.xs,  # type: ignore[union-attr]
-            )
-            optimal_nms = ScoreMetric(
-                name=actual_f_measure.best_nms_threshold.name,  # type: ignore[union-attr]
-                value=actual_f_measure.best_nms_threshold.value,  # type: ignore[union-attr]
-            )
             return [
                 generate_expected_default_dashboard_metric_groups(actual_f_measure)[0],
                 LineMetricsGroup(
-                    metrics=[per_confidence],
+                    metrics=[
+                        cast(CurveMetric, actual_f_measure.f_measure_per_confidence)
+                    ],
                     visualization_info=LineChartInfo(
                         name="F-measure per confidence",
                         x_axis_label="Confidence threshold",
@@ -1825,13 +1805,15 @@ class TestFMeasure:
                     ),
                 ),
                 TextMetricsGroup(
-                    metrics=[optimal_confidence],
+                    metrics=[
+                        cast(ScoreMetric, actual_f_measure.best_confidence_threshold)
+                    ],
                     visualization_info=TextChartInfo(
                         name="Optimal confidence threshold"
                     ),
                 ),
                 LineMetricsGroup(
-                    metrics=[per_nms],
+                    metrics=[cast(CurveMetric, actual_f_measure.f_measure_per_nms)],
                     visualization_info=LineChartInfo(
                         name="F-measure per nms",
                         x_axis_label="NMS threshold",
@@ -1839,7 +1821,7 @@ class TestFMeasure:
                     ),
                 ),
                 TextMetricsGroup(
-                    metrics=[optimal_nms],
+                    metrics=[cast(ScoreMetric, actual_f_measure.best_nms_threshold)],
                     visualization_info=TextChartInfo(name="Optimal nms threshold"),
                 ),
             ]

--- a/ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py
+++ b/ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py
@@ -1797,22 +1797,22 @@ class TestFMeasure:
             actual_f_measure: FMeasure,
         ):
             per_confidence = CurveMetric(
-                name=actual_f_measure.f_measure_per_confidence.name,
-                ys=actual_f_measure.f_measure_per_confidence.ys,
-                xs=actual_f_measure.f_measure_per_confidence.xs,
+                name=actual_f_measure.f_measure_per_confidence.name,  # type: ignore[union-attr]
+                ys=actual_f_measure.f_measure_per_confidence.ys,  # type: ignore[union-attr]
+                xs=actual_f_measure.f_measure_per_confidence.xs,  # type: ignore[union-attr]
             )
             optimal_confidence = ScoreMetric(
-                name=actual_f_measure.best_confidence_threshold.name,
-                value=actual_f_measure.best_confidence_threshold.value,
+                name=actual_f_measure.best_confidence_threshold.name,  # type: ignore[union-attr]
+                value=actual_f_measure.best_confidence_threshold.value,  # type: ignore[union-attr]
             )
             per_nms = CurveMetric(
-                name=actual_f_measure.f_measure_per_nms.name,
-                ys=actual_f_measure.f_measure_per_nms.ys,
-                xs=actual_f_measure.f_measure_per_nms.xs,
+                name=actual_f_measure.f_measure_per_nms.name,  # type: ignore[union-attr]
+                ys=actual_f_measure.f_measure_per_nms.ys,  # type: ignore[union-attr]
+                xs=actual_f_measure.f_measure_per_nms.xs,  # type: ignore[union-attr]
             )
             optimal_nms = ScoreMetric(
-                name=actual_f_measure.best_nms_threshold.name,
-                value=actual_f_measure.best_nms_threshold.value,
+                name=actual_f_measure.best_nms_threshold.name,  # type: ignore[union-attr]
+                value=actual_f_measure.best_nms_threshold.value,  # type: ignore[union-attr]
             )
             return [
                 generate_expected_default_dashboard_metric_groups(actual_f_measure)[0],

--- a/ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py
+++ b/ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py
@@ -1715,17 +1715,17 @@ class TestFMeasure:
         specified optional parameters
         """
 
-        def check_performance(performance, expected_score, expected_metrics):
+        def check_performance(performance, expected_score, expected_metric_groups):
             assert isinstance(performance, Performance)
             assert isinstance(performance.score, ScoreMetric)
             assert performance.score.name == "f-measure"
             assert performance.score.value == pytest.approx(expected_score)
             # Checking dashboard metrics
-            for expected_metric in expected_metrics:
-                metric_index = expected_metrics.index(expected_metric)
-                actual_metric_group = performance.dashboard_metrics[metric_index]
-                if isinstance(expected_metric, BarMetricsGroup):
-                    assert actual_metric_group.metrics == expected_metric.metrics
+            for expected_metric_group in expected_metric_groups:
+                metric_group_index = expected_metric_groups.index(expected_metric_group)
+                actual_metric_group = performance.dashboard_metrics[metric_group_index]
+                if isinstance(expected_metric_group, BarMetricsGroup):
+                    assert actual_metric_group.metrics == expected_metric_group.metrics
                     assert isinstance(
                         actual_metric_group.visualization_info, BarChartInfo
                     )
@@ -1741,11 +1741,16 @@ class TestFMeasure:
                         actual_metric_group.visualization_info.type
                         == VisualizationType.RADIAL_BAR
                     )
-                if isinstance(expected_metric, LineMetricsGroup):
-                    assert actual_metric_group.metrics == expected_metric.metrics
+                if isinstance(expected_metric_group, LineMetricsGroup):
+                    assert len(actual_metric_group.metrics) == 1
+                    expected_metric = expected_metric_group.metrics[0]
+                    actual_metric = actual_metric_group.metrics[0]
+                    assert actual_metric.name == expected_metric.name
+                    assert actual_metric.ys == expected_metric.ys
+                    assert actual_metric.xs == expected_metric.xs
                     assert (
                         actual_metric_group.visualization_info.name
-                        == expected_metric.visualization_info.name
+                        == expected_metric_group.visualization_info.name
                     )
                     assert (
                         actual_metric_group.visualization_info.palette
@@ -1753,17 +1758,17 @@ class TestFMeasure:
                     )
                     assert (
                         actual_metric_group.visualization_info.x_axis_label
-                        == expected_metric.visualization_info.x_axis_label
+                        == expected_metric_group.visualization_info.x_axis_label
                     )
                     assert (
                         actual_metric_group.visualization_info.y_axis_label
-                        == expected_metric.visualization_info.y_axis_label
+                        == expected_metric_group.visualization_info.y_axis_label
                     )
-                if isinstance(expected_metric, TextMetricsGroup):
-                    assert actual_metric_group.metrics == expected_metric.metrics
+                if isinstance(expected_metric_group, TextMetricsGroup):
+                    assert actual_metric_group.metrics == expected_metric_group.metrics
                     assert (
                         actual_metric_group.visualization_info.name
-                        == expected_metric.visualization_info.name
+                        == expected_metric_group.visualization_info.name
                     )
                     assert (
                         actual_metric_group.visualization_info.palette
@@ -1774,7 +1779,9 @@ class TestFMeasure:
                         == VisualizationType.TEXT
                     )
 
-        def generate_expected_default_dashboard_metrics(actual_f_measure: FMeasure):
+        def generate_expected_default_dashboard_metric_groups(
+            actual_f_measure: FMeasure,
+        ):
             return [
                 BarMetricsGroup(
                     metrics=list(actual_f_measure.f_measure_per_label.values()),
@@ -1786,11 +1793,31 @@ class TestFMeasure:
                 )
             ]
 
-        def generate_expected_optional_dashboard_metrics(actual_f_measure: FMeasure):
+        def generate_expected_optional_dashboard_metric_groups(
+            actual_f_measure: FMeasure,
+        ):
+            per_confidence = CurveMetric(
+                name=actual_f_measure.f_measure_per_confidence.name,
+                ys=actual_f_measure.f_measure_per_confidence.ys,
+                xs=actual_f_measure.f_measure_per_confidence.xs,
+            )
+            optimal_confidence = ScoreMetric(
+                name=actual_f_measure.best_confidence_threshold.name,
+                value=actual_f_measure.best_confidence_threshold.value,
+            )
+            per_nms = CurveMetric(
+                name=actual_f_measure.f_measure_per_nms.name,
+                ys=actual_f_measure.f_measure_per_nms.ys,
+                xs=actual_f_measure.f_measure_per_nms.xs,
+            )
+            optimal_nms = ScoreMetric(
+                name=actual_f_measure.best_nms_threshold.name,
+                value=actual_f_measure.best_nms_threshold.value,
+            )
             return [
-                generate_expected_default_dashboard_metrics(actual_f_measure),
+                generate_expected_default_dashboard_metric_groups(actual_f_measure)[0],
                 LineMetricsGroup(
-                    metrics=[actual_f_measure.f_measure_per_confidence],
+                    metrics=[per_confidence],
                     visualization_info=LineChartInfo(
                         name="F-measure per confidence",
                         x_axis_label="Confidence threshold",
@@ -1798,13 +1825,13 @@ class TestFMeasure:
                     ),
                 ),
                 TextMetricsGroup(
-                    metrics=[actual_f_measure.best_confidence_threshold],
+                    metrics=[optimal_confidence],
                     visualization_info=TextChartInfo(
                         name="Optimal confidence threshold"
                     ),
                 ),
                 LineMetricsGroup(
-                    metrics=[actual_f_measure.f_measure_per_nms],
+                    metrics=[per_nms],
                     visualization_info=LineChartInfo(
                         name="F-measure per nms",
                         x_axis_label="NMS threshold",
@@ -1812,7 +1839,7 @@ class TestFMeasure:
                     ),
                 ),
                 TextMetricsGroup(
-                    metrics=[actual_f_measure.best_nms_threshold],
+                    metrics=[optimal_nms],
                     visualization_info=TextChartInfo(name="Optimal nms threshold"),
                 ),
             ]
@@ -1827,14 +1854,14 @@ class TestFMeasure:
         # Checking "Performance" object returned by "get_performance" for "FMeasure" object initialized with default
         # optional parameters
         f_measure = FMeasure(result_set)
-        expected_dashboard_metrics = generate_expected_default_dashboard_metrics(
-            f_measure
+        expected_dashboard_metric_groups = (
+            generate_expected_default_dashboard_metric_groups(f_measure)
         )
         actual_performance = f_measure.get_performance()
         check_performance(
             performance=actual_performance,
             expected_score=0.2857142857142856,
-            expected_metrics=expected_dashboard_metrics,
+            expected_metric_groups=expected_dashboard_metric_groups,
         )
         # Check for incorrect prediction dataset
         incorrect_prediction_dataset = self.incorrect_prediction_dataset()
@@ -1844,14 +1871,14 @@ class TestFMeasure:
             prediction_dataset=incorrect_prediction_dataset,
         )
         f_measure = FMeasure(incorrect_result_set)
-        expected_dashboard_metrics = generate_expected_default_dashboard_metrics(
-            f_measure
+        expected_dashboard_metric_groups = (
+            generate_expected_default_dashboard_metric_groups(f_measure)
         )
         actual_performance = f_measure.get_performance()
         check_performance(
             performance=actual_performance,
             expected_score=0.15384615384615372,
-            expected_metrics=expected_dashboard_metrics,
+            expected_metric_groups=expected_dashboard_metric_groups,
         )
         # Checking attributes of "FMeasure" class object initialized with specified values of optional parameters
         f_measure = FMeasure(
@@ -1860,14 +1887,14 @@ class TestFMeasure:
             vary_nms_threshold=True,
             cross_class_nms=True,
         )
-        expected_dashboard_metrics = generate_expected_optional_dashboard_metrics(
-            f_measure
+        expected_dashboard_metric_groups = (
+            generate_expected_optional_dashboard_metric_groups(f_measure)
         )
         actual_performance = f_measure.get_performance()
         check_performance(
             performance=actual_performance,
             expected_score=0.2857142857142856,
-            expected_metrics=expected_dashboard_metrics,
+            expected_metric_groups=expected_dashboard_metric_groups,
         )
         # Check for incorrect prediction dataset
         f_measure = FMeasure(
@@ -1876,12 +1903,12 @@ class TestFMeasure:
             vary_nms_threshold=True,
             cross_class_nms=True,
         )
-        expected_dashboard_metrics = generate_expected_optional_dashboard_metrics(
-            f_measure
+        expected_dashboard_metric_groups = (
+            generate_expected_optional_dashboard_metric_groups(f_measure)
         )
         actual_performance = f_measure.get_performance()
         check_performance(
             performance=actual_performance,
             expected_score=0.15384615384615372,
-            expected_metrics=expected_dashboard_metrics,
+            expected_metric_groups=expected_dashboard_metric_groups,
         )

